lib/resourcemerge/core.go

```diff
@@ -90,7 +90,8 @@ func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.Pod
 	setInt32Ptr(modified, &existing.Priority, required.Priority)
 }
 
-func ensureContainer(modified *bool, existing *corev1.Container, required corev1.Container) {
+func ensureContainer(modified *bool, existing *corev1.Container, required corev1.Container) bool {
+	overlap := false
 	resourcemerge.SetStringIfSet(modified, &existing.Name, required.Name)
```

Comment:
The latest tests apparently panicked on this line, looks like the... existing container didn't exist?